modules/data.land/R/validate_events.R

```diff
@@ -0,0 +1,53 @@
+#' Validate PEcAn events JSON against schema v0.1.0
+#'
+#' Validates a PEcAn events JSON file (single-site object or an array of site
+#' objects) against the bundled JSON Schema (draft 2020-12) using the AJV
+#' engine.
+#'
+#' - Logs an error and returns FALSE if the JSON file does not exist or does
+#'   not conform to the schema.
+#' - Logs a warning and returns TRUE if the optional package `jsonvalidate` is
+#'   not installed, so calling code can proceed without a hard dependency.
+#'
+#' @param events_json character. Path to the JSON file to validate.
+#' @param verbose logical. When `TRUE`, include detailed AJV messages on error.
+#'
+#' @return Logical TRUE if valid (or validator unavailable), FALSE if invalid.
+#'
+#' @author David LeBauer
+#'
+#' @examples
+#' # validate_events_json(system.file("events_fixtures/events_site1.json",
+#' #                                package = "PEcAn.data.land"))
+#'
+#' @export
+validate_events_json <- function(events_json, verbose = TRUE) {
+  if (!file.exists(events_json)) {
+    PEcAn.logger::logger.error(glue::glue("events_json file does not exist: {events_json}"))
+    return(FALSE)
+  }
+
+  if (!requireNamespace("jsonvalidate", quietly = TRUE)) {
+    PEcAn.logger::logger.warn("Skipping events schema validation: package 'jsonvalidate' not installed.")
+    return(TRUE)
```

Comment:
I see the appeal of allowing work to proceed if validation isn't available, but I'm not wild about returning TRUE -- seems too easy to interpret as "the file has been checked and is OK."  I'd be OK with any of returning FALSE ("The file has not been validated to be OK, in this case because we couldn't validate it"), returning NA  and letting callers decide whether to stop or continue with an unchecked file, or changing design to throw an error if jsonvalidate isn't installed (in which case it would still be fine to leave jsonvalidate in Suggests -- it's optional for anyone who uses data.land for things other than this function)

Reply:
Of these I think the NA approach best, so here's a suggestion for it. 
```suggestion
    PEcAn.logger::logger.warn("Skipping events schema validation: package 'jsonvalidate' not installed.")
    return(NA)
```